mantidproject/mantid
# Clean internal variables inside SmoothNeighbours

**Description of work.**

The first 3 commits are cleanup, the final is a release note. The real fix is inside commit 4. Please see that commit message for more details.

**To test:**

The following script should so the memory usage go down after the `DeleteWorkspace` call. Before this fix it did not.

```python
"""SmoothNeighbours appears to hold on to memory when replacing a workspace with
itself in v6.3
"""
import psutil

MB = 1. / (1024*1024)

def pprint_memory_stats(process: psutil.Process):
    mem_info = process.memory_info()
    print(f"Resident Memory (this process): {mem_info.rss*MB:.3f}MB")
    return mem_info.rss*MB

proc = psutil.Process()
print("Starting...")
pprint_memory_stats(proc)

print("Importing mantid")
from mantid.simpleapi import Load, SmoothNeighbours, DeleteWorkspace

SXD23767 = Load(Filename='~/Data/TrainingCourseData/SXD23767.raw', LoadMonitors='Exclude')
rss = [pprint_memory_stats(proc)]

SXD23767 = SmoothNeighbours(SXD23767)
rss.append(pprint_memory_stats(proc))
print(f"RSS Diff: {(rss[-1] - rss[-2]):.3f}MB")
DeleteWorkspace(SXD23767)
rss.append(pprint_memory_stats(proc))
print(f"RSS Diff: {(rss[-1] - rss[-2]):.3f}MB")
```

Fixes #33661 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
